src/content/guides/packages.md

```diff
@@ -261,6 +261,184 @@ To update `pubspec.lock` run `dart pub get` without `--enforce-lockfile`.
+/
+  packages/
+    shared/
+      pubspec.yaml
+      pubspec.lock
+      .dart_tool/package_config.json
+    client_package/
+      pubspec.yaml
+      pubspec.lock
+      .dart_tool/package_config.json
+    server_package/
+      pubspec.yaml
+      pubspec.lock
+      .dart_tool/package_config.json
+```
+
+There are some downsides to this setup:
+
+* You need to run `dart pub get` once for each package.
+* You risk ending up with different versions of dependencies for each package.
+  Leading to confusion when context switching between the packages.
+* If you open the root folder in your IDE, the dart analyzer will have to have a
+  separate analysis contexts for each package (increasing memory usage).
+
+Pub allows you to organize your repository as a "workspace" using a single
+shared resolution for all your packages.
+
+:::note
+Using a single shared dependency resolution for all your packages increases
+the risks of dependency conflicts. Because Dart does not allow multiple versions
+of the same package.
+
+If the packages are going to be used together (as is commonly the case),
+this is often a feature, as it forces you to resolve incompatibilities between your
+packages when they arise, rather then when you start using the packages.
+:::
+
+To create a workspace:
+
+* Add a `pubspec.yaml` at the repository root directory with a `workspace` entry
+  enummerating the paths to the packages of the respository (the workspace
+  packages):
+
+```yaml
+name: _
+publish_to: none
+environment:
+  sdk: ^3.6.0
+workspace:
+  - packages/helper
+  - packages/client_package
+  - packages/server_package
+```
+
+* For each of the existing `pubspec.yaml`s, make sure their SDK constraint is at
+  least `^3.6.0` and add a `resolution` entry:
+
+  ```yaml
+  environment:
+    sdk: ^3.6.0
+  resolution: workspace
+  ```
+
+* Run `dart pub get` anywhere in the repository, and this will:
+  * Create a single `pubspec.lock` next to the root `pubspec.yaml` that contains
+    the a resolution of all the `dependencies` and `dev_dependencies` of all the
+    workspace packages. 
+  * Create a single shared `.dart_tool/package_config.json` that maps package
+    names to file locations.
+  * Delete any other existing `pubspec.lock` and
+    `.dart_tool/package_config.json` files next to workspace packages.
+
+Now the file structure looks like this:
+
+```console
+/
+  packages/
+    shared/
+      pubspec.yaml
+    client_package/
+      pubspec.yaml
+    server_package/
+      pubspec.yaml
+  pubspec.yaml
+  pubspec.lock
+  .dart_tool/package_config.json
+```
+
+:::version-note
+Pub workspaces was introduced in Dart 3.6.0.
+
+To use pub workspaces, all your workspace packages (but not your dependencies)
+must have a SDK version constraint of `^3.6.0` or higher.
+:::
+
+### Interdependencies between workspace packages
+
+If any of the workspace packages depend on each other, they will automatically
+resolve to the one in the workspace, regardless of the source.
```

Comment:
Can you explain this differently? It sounds confusing to me "If _any of the workspace packages_ depend _on each other_..." makes it sounds like there's packages A and B and in the workspace, and A depends on B and B depends on A. But then the resolution is "...they will automatically resolve to the one in the workspace..." -- I thought they are both in the workspace though, right? Which resolves to which? 

I understand the example given in rest of this section, but this introduction sounds like it's talking about a different situation, maybe? Hope it's not too much trouble to explain again, thank you!